pkg/cloud/instance.go

```diff
@@ -239,5 +238,18 @@ func (c *client) DestroyVMInstance(csMachine *infrav1.CloudStackMachine) error {
 	} else if err != nil {
 		return err
 	}
+
+	if err := c.ResolveVMInstanceDetails(csMachine); err == nil && csMachine.Status.InstanceState == "Expunging" ||
```

Comment:
Don't we need () around `csMachine.Status.InstanceState == "Expunging" || csMachine.Status.InstanceState == "Expunged"`

Reply:
Yes, good catch!